Topics that match more than one Template Subscription break the Topic View page

If a Topic becomes recommended more than once, the Topic View page (where you could either Subscribe or not to the recommended Topic) cannot be displayed due to an expectation of a singleOpt.

```
2015-07-22T20:28:46.418646+00:00 app[web.1]: ! @6mphldi47 - Internal server error, for (GET) [/topic/6548] ->
2015-07-22T20:28:46.418647+00:00 app[web.1]:
2015-07-22T20:28:46.418651+00:00 app[web.1]: play.api.UnexpectedException: Unexpected exception[RuntimeException: SqlMappingError(too many rows when expecting a single one)]
2015-07-22T20:28:46.418653+00:00 app[web.1]: 	at play.api.http.HttpErrorHandlerExceptions$.throwableToUsefulException(HttpErrorHandler.scala:261) ~[com.typesafe.play.play_2.11-2.4.2.jar:2.4.2]
2015-07-22T20:28:46.418655+00:00 app[web.1]: 	at play.api.http.DefaultHttpErrorHandler.onServerError(HttpErrorHandler.scala:191) ~[com.typesafe.play.play_2.11-2.4.2.jar:2.4.2]
2015-07-22T20:28:46.418657+00:00 app[web.1]: 	at play.api.GlobalSettings$class.onError(GlobalSettings.scala:179) [com.typesafe.play.play_2.11-2.4.2.jar:2.4.2]
2015-07-22T20:28:46.418659+00:00 app[web.1]: 	at play.api.DefaultGlobal$.onError(GlobalSettings.scala:212) [com.typesafe.play.play_2.11-2.4.2.jar:2.4.2]
2015-07-22T20:28:46.418661+00:00 app[web.1]: 	at play.api.http.GlobalSettingsHttpErrorHandler.onServerError(HttpErrorHandler.scala:94) [com.typesafe.play.play_2.11-2.4.2.jar:2.4.2]
2015-07-22T20:28:46.418663+00:00 app[web.1]: 	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$9$$anonfun$apply$1.applyOrElse(PlayDefaultUpstreamHandler.scala:158) [com.typesafe.play.play-netty-server_2.11-2.4.2.jar:2.4.2]
2015-07-22T20:28:46.418665+00:00 app[web.1]: 	at play.core.server.netty.PlayDefaultUpstreamHandler$$anonfun$9$$anonfun$apply$1.applyOrElse(PlayDefaultUpstreamHandler.scala:155) [com.typesafe.play.play-netty-server_2.11-2.4.2.jar:2.4.2]
2015-07-22T20:28:46.418667+00:00 app[web.1]: 	at scala.runtime.AbstractPartialFunction.apply(AbstractPartialFunction.scala:36) [org.scala-lang.scala-library-2.11.6.jar:na]
2015-07-22T20:28:46.418669+00:00 app[web.1]: 	at scala.util.Failure$$anonfun$recover$1.apply(Try.scala:215) [org.scala-lang.scala-library-2.11.6.jar:na]
2015-07-22T20:28:46.418672+00:00 app[web.1]: 	at scala.util.Try$.apply(Try.scala:191) [org.scala-lang.scala-library-2.11.6.jar:na]
2015-07-22T20:28:46.418674+00:00 app[web.1]: Caused by: java.lang.RuntimeException: SqlMappingError(too many rows when expecting a single one)
2015-07-22T20:28:46.418677+00:00 app[web.1]: 	at scala.sys.package$.error(package.scala:27) ~[org.scala-lang.scala-library-2.11.6.jar:na]
2015-07-22T20:28:46.418680+00:00 app[web.1]: 	at anorm.SqlRequestError$class.toFailure(Anorm.scala:20) ~[com.typesafe.play.anorm_2.11-2.4.0.jar:2.4.0]
2015-07-22T20:28:46.418682+00:00 app[web.1]: 	at anorm.SqlMappingError.toFailure(Anorm.scala:38) ~[com.typesafe.play.anorm_2.11-2.4.0.jar:2.4.0]
2015-07-22T20:28:46.418684+00:00 app[web.1]: 	at anorm.Sql$$anonfun$asTry$2$$anonfun$apply$7.apply(Anorm.scala:303) ~[com.typesafe.play.anorm_2.11-2.4.0.jar:2.4.0]
2015-07-22T20:28:46.418686+00:00 app[web.1]: 	at anorm.Sql$$anonfun$asTry$2$$anonfun$apply$7.apply(Anorm.scala:303) ~[com.typesafe.play.anorm_2.11-2.4.0.jar:2.4.0]
2015-07-22T20:28:46.418689+00:00 app[web.1]: 	at anorm.SqlResult$class.fold(SqlResult.scala:23) ~[com.typesafe.play.anorm_2.11-2.4.0.jar:2.4.0]
2015-07-22T20:28:46.418691+00:00 app[web.1]: 	at anorm.Error.fold(SqlResult.scala:31) ~[com.typesafe.play.anorm_2.11-2.4.0.jar:2.4.0]
2015-07-22T20:28:46.418693+00:00 app[web.1]: 	at anorm.Sql$$anonfun$asTry$2.apply(Anorm.scala:303) ~[com.typesafe.play.anorm_2.11-2.4.0.jar:2.4.0]
2015-07-22T20:28:46.418695+00:00 app[web.1]: 	at anorm.Sql$$anonfun$asTry$2.apply(Anorm.scala:303) ~[com.typesafe.play.anorm_2.11-2.4.0.jar:2.4.0]
2015-07-22T20:28:46.418698+00:00 app[web.1]: 	at scala.util.Success.flatMap(Try.scala:230) ~[org.scala-lang.scala-library-2.11.6.jar:na]
```

--- a/app/workers/Conveyor.scala
+++ b/app/workers/Conveyor.scala
@@ -89,12 +89,24 @@ object Conveyor {
       }
     }
 
-    def subscribe(action: String) = mints.foreach( i =>
-      Subscription.create(i.subscriberId, topic.id, action, sub.created, new Date))
+    def subscribe(action: String) = {
+      mints.foreach( i =>
+        if(sub.subscribesTo(topic.id)) {
+          println("Conveyor: Subscriber already subscribes to Topic, not re-subscribing.")
+        } else {
+          println("Conveyor: Added new Subscription.")
+          Subscription.create(i.subscriberId, topic.id, action, sub.created, new Date)
+        })
+    }
 
     def review() = {
       mints.foreach( t =>
-        TopicPick.create(t.subscriberId, topic.id, conveyorAgent.id))
+        if(TopicPick.picked(topic.id, t.subscriberId)) {
+          println("Conveyor: TopicPick detected duplicate so did nothing.")
+        } else {
+          println("Conveyor: Added new TopicPick.")
+          TopicPick.create(t.subscriberId, topic.id, conveyorAgent.id)
+        })
     }
   }
 

